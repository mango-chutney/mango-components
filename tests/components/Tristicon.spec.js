// @flow

import 'raf/polyfill';
import 'jest-styled-components';
import React from 'react';
import renderer from 'react-test-renderer';
import MC from '../../src';

const Tristicon = MC.Tristicon.createComponent();

function TristiconComponent() {
  return <Tristicon icon="star" />;
}

it('Tristicon component renders correctly', () => {
  const tree = renderer.create(<TristiconComponent />).toJSON();
  expect(tree).toMatchSnapshot();
});
