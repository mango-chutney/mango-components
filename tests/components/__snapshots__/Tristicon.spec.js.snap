// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel components (Panel, PanelHeading, PanelSection) renders correctly 1`] = `
.c0 {
  height: 3.125rem;
  position: relative;
  width: 3.125rem;
}

.c1 {
  height: 100%;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-transform: rotateZ(-90deg);
  -ms-transform: rotateZ(-90deg);
  transform: rotateZ(-90deg);
  width: 100%;
}

.c1 svg {
  overflow: visible;
}

.c1 circle {
  -webkit-transition: stroke-dashoffset 0.5s linear;
  transition: stroke-dashoffset 0.5s linear;
}

.c2 {
  stroke: #e2e7ee;
}

.c3 {
  stroke: #2ea1f8;
  -webkit-animation: CYIfC 0.5s normal forwards;
  animation: CYIfC 0.5s normal forwards;
}

<div
  className="c0"
  width={undefined}
>
  <div
    className="c1"
  >
    <svg
      height="100%"
      viewBox="0 0 100 100"
      width="100%"
    >
      <circle
        className="c2"
        cx={50}
        cy={50}
        fill="none"
        r={40}
        strokeWidth={7}
      />
      <circle
        className="c3"
        cx={50}
        cy={50}
        fill="none"
        r={40}
        strokeDasharray={251}
        strokeDashoffset={248.49}
        strokeLinecap={undefined}
        strokeWidth={7}
      />
    </svg>
  </div>
</div>
`;

exports[`Tristicon component renders correctly 1`] = `
.c0::before {
  content: "";
  display: inline-block;
  font: normal normal normal 0.875rem tristicons;
  font-size: inherit;
  text-rendering: auto;
  vertical-align: inherit;
}

<i
  className="c0"
  icon={undefined}
/>
`;
